src/Symfony/Component/Security/Core/Authorization/Voter/VoterInterface.php

```diff
@@ -33,7 +33,7 @@ interface VoterInterface
      * @param mixed $subject    The subject to secure
      * @param array $attributes An array of attributes associated with the method being invoked
```

Comment:
Based on https://github.com/phpstan/phpstan-symfony/pull/291, this may be changed to `array<mixed>`

Reply:
Personally, I don't like PHPstan's requirement to make any `array` a generic (so requiring to specify at least one type). I think it's OK to use Psalm & PHPstorm's definition of `array = array<array-key, mixed>` in Symfony's code base and keep changes like this for a PHPstan stub.

Reply:
There are multiple places in the Symfony code where `array` is used and could be replaced by `string[]` or others more precise types. So currently, `array` just mean "We didn't precise, but maybe don't accept everything" and not "Every `array<array-key, mixed>` is allowed".

PHPStan strategy force to pass on every `array` to write instead, sometimes `string[]`, sometime `array<string, mixed>` and sometimes `array<mixed>`. The advantage of `array<mixed>` (or `mixed[]` if preferred) is the fact than we do know that the function really accept everything, it's supposed to be checked when changing `array` into `array<mixed>` (or `string[]` or ...).

I may be wrong but I don't feel like changing `array` into `array<mixed>` on some places would require a lot of extra-maintenance to the symfony/symfony project when it would require to duplicate a lot of classes/interface on phpstan/phpstan-symfony just to solve this. So it would be a nice gesture to Symfony+PHPStan users to consider using `array<mixed>`.